hil/cli/network.py

```diff
@@ -34,30 +35,81 @@ def network_delete(network):
 
 @network.command(name='show')
 @click.argument('network')
-def network_show(network):
+@click.option('--json', 'jsonout', is_flag=True)
+def network_show(network, jsonout):
     """Display information about network"""
-    q = client.network.show(network)
-    for item in q.items():
-        sys.stdout.write("%s\t  :  %s\n" % (item[0], item[1]))
+    raw_output = client.network.show(network)
+
+    if jsonout:
+        print_json(raw_output)
+
+    # Collect raw output into presentable form
+    channels = '\n'.join(raw_output['channels']) + '\n'
+    projects = '\n'.join(raw_output['access']) + '\n'
+
+    connected_nodes = ''
+    for connected_node, nics in raw_output['connected-nodes'].iteritems():
+        # Because a node can be connected to same network over multiple nics
+        all_nics = ','.join(nics)
+        connected_nodes += connected_node + ' (' + all_nics + ')\n'
+
+    # Make the table and print it
+    network_show_table = PrettyTable()
+    network_show_table.field_names = ['Field', 'Value']
+    network_show_table.add_row(['Name', raw_output['name']])
+    network_show_table.add_row(['Owner', raw_output['owner']])
+    network_show_table.add_row(['Channels', channels])
+    network_show_table.add_row(['Access', projects])
+    network_show_table.add_row(['Connected Nodes', connected_nodes.rstrip()])
+    print(network_show_table)
```

Comment:
I'm finding this kind of imperative construction less readable than it could be; the comment at the top of this block is a bit of a code smell. With an appropriate helper function that would let us do:

```python
print(make_table(
    field_names=['Field', 'Value'],
    rows=[
        ['Name', raw_output['name']],
        ['Owner', raw_output['owner']],
        # etc
    ]))
```

For the cases where we're currently building up the list of rows in a for loop, we could use a list comprehension.

I don't think it would be too hard to generalize the helpers you've already written to cover this.

Reply:
I think this is a cool idea too.  That way, the tables can all be generalized for future use, ensuring that our CLI looks uniform.

Reply:
Done. Let me know what you guys think.